src/components/molecules/table-fileds-filter/index.tsx

```diff
@@ -0,0 +1,200 @@
+  const [currentlySelected, setCurrentlySelected] = useState<string[]>([])
+
+  const onTriggerClick = () => {
+    setOpen(true)
+  }
+
+  const toggleCheck = (id: string) => {
+    if (currentlySelected.includes(id)) {
+      setCurrentlySelected(currentlySelected.filter((f) => f !== id))
+    } else {
+      setCurrentlySelected([...currentlySelected, id])
+    }
+  }
+
+  useEffect(() => {
+    if (open) {
+      setCurrentlySelected(selectedFields)
+    }
+  }, [open, selectedFields])
+
+  useEffect(() => {
+    if (!open) {
+      onBlur(currentlySelected)
+    }
+  }, [open])
+
+  // close dropdown "manually" on click outside the menu
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (!contentRef.current?.contains(event.target)) {
+        setOpen(false)
+      }
+    }
+    document.addEventListener("mousedown", handleClickOutside)
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside)
+    }
+  }, [contentRef])
+
+  return (
+    <DropdownMenu.Root open={open}>
+      <DropdownMenu.Trigger>
+        <Button
+          onClick={onTriggerClick}
+          variant="secondary"
+          className="rounded-lg h-[32px] px-3 text-small font-semibold text-grey-90 inline-flex"
+        >
+          <span className="flex whitespace-nowrap items-center gap-1">
+            Add fields <PlusIcon size={14} />
+          </span>
+        </Button>
+      </DropdownMenu.Trigger>
+
+      <DropdownMenu.Content
+        ref={contentRef}
+        className="w-[240px] bg-white shadow rounded-xl p-2"
+      >
+        {fields.map((f) => (
+          <FieldMenuItem
+            key={f.id}
+            field={f}
+            onChange={() => toggleCheck(f.id)}
+            checked={currentlySelected.includes(f.id)}
+          />
+        ))}
+      </DropdownMenu.Content>
+    </DropdownMenu.Root>
+  )
+}
+
+/**
+ * Table fields filter root container.
+ */
+function TableFieldsFilters(props: TableFieldsFilterProps) {
+  const { fields, onChange } = props
+
+  const [selectedFields, setSelectedFields] = useState<Field["id"][]>([])
+
+  useEffect(() => {
+    onChange(selectedFields)
+  }, [selectedFields])
+
+  const removeSelected = (id: string) => {
+    setSelectedFields(selectedFields.filter((f) => f !== id))
+  }
+
+  const _selected = [...selectedFields]
+  _selected.sort((a, b) => a.localeCompare(b))
+
+  const visibleFields = _selected.map((id) => fields.find((f) => f.id === id))
+
+  return (
+    <>
+      <span className="text-small font-semibold whitespace-nowrap text-gray-500 mr-2">
+        Currently editing these fields:
+      </span>
+
+      {visibleFields.map((f) => (
+        <Chip key={f!.id} remove={() => removeSelected(f!.id)} {...f!} />
+      ))}
+
+      <FieldsMenu
+        fields={sortBy(fields, "id")}
+        onBlur={setSelectedFields}
+        selectedFields={_selected}
+      />
+    </>
```

Comment:
```suggestion
    <div className="flex-wrap flex items-center gap-y-2">
      <span className="text-small font-semibold whitespace-nowrap text-gray-500 mr-2">
        Currently editing these fields:
      </span>

      {visibleFields.map((f) => (
        <Chip key={f!.id} remove={() => removeSelected(f!.id)} {...f!} />
      ))}

      <FieldsMenu
        fields={sortBy(fields, "id")}
        onBlur={setSelectedFields}
        selectedFields={_selected}
      />
    </div>
```

nit: adding some vertical spacing between "rows" when wrapped, like in the design.